### ISSUE TITLE
Test failure: profiler/eventpipe/eventpipe/eventpipe.sh

### ISSUE THREAD

**Issue by v-haren** [dotnet/runtime#36195]
failed in job: [runtime-coreclr outerloop 20200510.5 ](https://dev.azure.com/dnceng/public/_build/results?buildId=638906&view=ms.vss-test-web.build-test-results-tab&runId=19860234&resultId=100000&paneView=debug)

Error message
~~~
Assert failure(PID 58 [0x0000003a], Thread: 69 [0x0045]): !"Hit an error in the diagnostic server thread
."
 File: /__w/1/s/src/coreclr/src/vm/diagnosticserver.cpp Line: 99
 Image: /root/helix/work/correlation/corerun

/root/helix/work/workitem/profiler/eventpipe/eventpipe/eventpipe.sh: line 312: 49 Segmentation fault $LAUNCHER $ExePath "${CLRTestExecutionArguments[@]}"

Return code: 1
Raw output file: /root/helix/work/workitem/profiler/eventpipe/Reports/profiler.eventpipe/eventpipe/eventpipe.output.txt
Raw output:
BEGIN EXECUTION
in takeLock
Microsoft (R) CoreCLR Native Image Generator - Version 5.0.0-ci
Copyright (c) Microsoft Corporation. All rights reserved.

Native image eventpipe.dll generated successfully.

in ReleaseLock
in takeLock
/root/helix/work/correlation/corerun eventpipe.dll ''
Profiler path: /root/helix/work/workitem/profiler/eventpipe/eventpipe/libProfiler.so

Profiler path: /root/helix/work/workitem/profiler/eventpipe/eventpipe/libProfiler.so
Profiler.dll!DllGetClassObject
Profiler.dll!Profiler::Initialize
This method being jitted should trigger events firing...

TriggerMethod found! Sending event...
Writing minidump to file /home/helixbot/dotnetbuild/dumps/coredump.58.dmp
Written 33157120 bytes (8095 pages) to core file
Dump successfully written

Test failed: Profiler tests are expected to contain the text 'PROFILER TEST PASSES' in the console output of the profilee app to indicate a passing test. Usually it is printed from the Shutdown() method of the profiler implementation. This text was not found in the output above.
Writing minidump to file /home/helixbot/dotnetbuild/dumps/coredump.49.dmp
Written 26755072 bytes (6532 pages) to core file
Dump successfully written
in ReleaseLock
Expected: 100
Actual: 139
END EXECUTION - FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=/root/helix/work/correlation
> /root/helix/work/workitem/profiler/eventpipe/eventpipe/eventpipe.sh
Expected: True
Actual: False


Stack trace
   at profiler_eventpipe._eventpipe_eventpipe_._eventpipe_eventpipe_sh() in /__w/1/s/artifacts/tests/coreclr/Linux.x64.Checked/TestWrappers/profiler.eventpipe/profiler.eventpipe.XUnitWrapper.cs:line 140
~~~

**Comment by tommcdon** [comment]
@josalem @davmason

**Comment by davmason** [comment]
@josalem I assume you are handling this because it's in the server code. If you need me to take a look let me know.

**Comment by josalem** [comment]
Got side tracked triaging some other issues this morning.  I need to get my hands on the dump which should show what happened.  I'll sync with you offline to chat about what this profiler test is doing.

**Comment by josalem** [comment]
Did an investigation yesterday.  After some pain, I managed to get a working `lldb` + `dotnet-sos` environment setup to look at the dumps.  @davmason was unable to repro the failure locally.  Given that this failure appears to be an isolated event so far, it is most likely difficult to reproduce on command.

I'm documenting my findings below in case this issue reoccurs.

When looking at the dumps that came along with the logs I found one thread to be deep inside the JIT and many of the threads to have their stacks truncated to the musl linker, presumably calling into the musl libc or another so.  I couldn't find symbols for the R2R'd DLLs in the pipeline artifacts, so anything that got R2R'd was missing source line and method names.  It looks like one thread was attempting to connect to the Diagnostics Server:

```
(lldb) clrstack -f
OS Thread Id: 0x3a (1)
        Child SP               IP Call Site
00007FFEE2964AE8 00007F87F531447A ld-musl-x86_64.so.1 + -1
00007FFEE2964C10                  [InlinedCallFrame: 00007ffee2964c10] System.Net.Sockets.ni.dll!/root/helix/work/correlation/System.Net.Sockets.ni.dll!Unknown
00007FFEE2964C10                  [InlinedCallFrame: 00007ffee2964c10] System.Net.Sockets.ni.dll!/root/helix/work/correlation/System.Net.Sockets.ni.dll!Unknown
00007FFEE2964C00 00007F877B4091C8 System.Diagnostics.Process.ni.dll!ILStubClass.IL_STUB_PInvoke + 152
00007FFEE2964CA0 00007F877CA0AC14 System.Net.Sockets.ni.dll!/root/helix/work/correlation/System.Net.Sockets.ni.dll!Unknown + 308
00007FFEE2964D70 00007F877CA0C57B System.Net.Sockets.ni.dll!/root/helix/work/correlation/System.Net.Sockets.ni.dll!Unknown + 219
00007FFEE2964DF0 00007F877CA05CAC System.Net.Sockets.ni.dll!/root/helix/work/correlation/System.Net.Sockets.ni.dll!Unknown + 188
00007FFEE2964E90 00007F877CA0D72B System.Net.Sockets.ni.dll!/root/helix/work/correlation/System.Net.Sockets.ni.dll!Unknown + 235
00007FFEE2964F50 00007F877C9EA48D System.Net.Sockets.ni.dll!/root/helix/work/correlation/System.Net.Sockets.ni.dll!Unknown + 397
00007FFEE2964FB0 00007F877C9E37C9 System.Net.Sockets.ni.dll!/root/helix/work/correlation/System.Net.Sockets.ni.dll!Unknown + 169
00007FFEE2965000 00007F877AF83750 System.Private.CoreLib.dll!/root/helix/work/correlation/System.Private.CoreLib.dll!Unknown + 96
00007FFEE2965030 00007F877B7B752A Microsoft.Diagnostics.NETCore.Client.ni.dll!/root/helix/work/correlation/Microsoft.Diagnostics.NETCore.Client.ni.dll!Unknown + 106 [/_/src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcHeader.cs @ 58]
00007FFEE2965050 00007F877B7B7990 Microsoft.Diagnostics.NETCore.Client.ni.dll!/root/helix/work/correlation/Microsoft.Diagnostics.NETCore.Client.ni.dll!Unknown + 80 [/_/src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcMessage.cs @ 102]
00007FFEE2965090 00007F877B7B700C Microsoft.Diagnostics.NETCore.Client.ni.dll!/root/helix/work/correlation/Microsoft.Diagnostics.NETCore.Client.ni.dll!Unknown + 76 [/_/src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcClient.cs @ 88]
00007FFEE29650C0 00007F877B7B67D3 Microsoft.Diagnostics.NETCore.Client.ni.dll!/root/helix/work/correlation/Microsoft.Diagnostics.NETCore.Client.ni.dll!Unknown + 163 [/_/src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsClient/EventPipeSession.cs @ 54]
00007FFEE29650F0 00007F877B61365D eventpipe.dll!/root/helix/work/workitem/profiler/eventpipe/eventpipe/eventpipe.dll!Unknown + 381 [/Users/runner/runners/2.166.4/work/1/s/src/coreclr/tests/src/profiler/eventpipe/eventpipe.cs @ 93]
00007FFEE2965130 00007F877B613469 eventpipe.dll!/root/helix/work/workitem/profiler/eventpipe/eventpipe/eventpipe.dll!Unknown + 73 [/Users/runner/runners/2.166.4/work/1/s/src/coreclr/tests/src/profiler/eventpipe/eventpipe.cs @ 28]
```

This is the managed code attempting to do something with the Unix Domain Socket for the diagnostics server and specifically it is making an EventPipe related call.  Based on the limited source line info, we can see this is a `StopTracing` command ([EventPipeSession @ line 54](https://github.com/dotnet/diagnostics/blob/55d5a20964f3ec990c6fe97b607161cda1a191e2/src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsClient/EventPipeSession.cs#L49-L54)).  Specifically, it is at the point where it is *waiting on a response* from the server.


The thread that was in the JIT was compiling an interop call to `Crypto::EvpMdSize` for `System.Security.Cryptography.Algorithms` as part of parsing the stream from EventPipe.  We can see TraceEvent in the stack without symbols, but based on the calls to FastSerializer we're presumably parsing the stream.  This thread is most likely the thread kicked off [here](https://github.com/dotnet/runtime/blob/ba453bd90ccce9848ce95651f0e13413660c09ee/src/coreclr/tests/src/profiler/eventpipe/eventpipe.cs#L85-L88) in the test:
```
(lldb) clrstack -f
OS Thread Id: 0x47 (10)
        Child SP               IP Call Site
00007F87EFE41AB8 00007F87F32B1480 libclrjit.so!ValueNumStore::VNNormalValue(ValueNumPair, ValueNumKind) at /__w/1/s/src/coreclr/src/jit/valuenum.cpp:1372
00007F87EFE41AC0 00007F87F30D16AF libclrjit.so!Compiler::optCreateAssertion(GenTree*, GenTree*, Compiler::optAssertionKind, bool) + 3519 at /__w/1/s/src/coreclr/src/jit/assertionprop.cpp:1032
00007F87EFE41AC0 00007F87F30D16A5 libclrjit.so!Compiler::optCreateAssertion(GenTree*, GenTree*, Compiler::optAssertionKind, bool) + 3509 at /__w/1/s/src/coreclr/src/jit/valuenum.h:425
00007F87EFE41B80 00007F87F30D36D3 libclrjit.so!Compiler::optAssertionGen(GenTree*) + 115 at /__w/1/s/src/coreclr/src/jit/gentree.h:171
00007F87EFE41BC0 00007F87F3240E39 libclrjit.so!Compiler::fgMorphTreeDone(GenTree*, GenTree*, int) + 441 at /__w/1/s/src/coreclr/src/jit/morph.cpp:15450
00007F87EFE41C00 00007F87F323C9CC libclrjit.so!Compiler::fgMorphTree(GenTree*, Compiler::MorphAddrContext*) + 716 at /__w/1/s/src/coreclr/src/jit/morph.cpp:15269
00007F87EFE41C40 00007F87F32658B7 libclrjit.so!Compiler::fgMorphStmts(BasicBlock*, bool*, bool*) + 359 at /__w/1/s/src/coreclr/src/jit/morph.cpp:15999
00007F87EFE41CA0 00007F87F32668D0 libclrjit.so!Compiler::fgMorphBlocks() + 336 at /__w/1/s/src/coreclr/src/jit/morph.cpp:16242
00007F87EFE41CF0 00007F87F310C5A2 libclrjit.so!ActionPhase<Compiler::compCompile(void**, unsigned int*, JitFlags*)::$_6>::DoPhase() + 18 at /__w/1/s/src/coreclr/src/jit/compiler.cpp:4530
00007F87EFE41CF0 00007F87F310C59A libclrjit.so!ActionPhase<Compiler::compCompile(void**, unsigned int*, JitFlags*)::$_6>::DoPhase() + 10 at /__w/1/s/src/coreclr/src/jit/compiler.cpp:4527
00007F87EFE41D10 00007F87F328CEE6 libclrjit.so!Phase::Run() + 118 at /__w/1/s/src/coreclr/src/jit/phase.cpp:62
00007F87EFE41D60 00007F87F310446E libclrjit.so!Compiler::compCompile(void**, unsigned int*, JitFlags*) + 2750 at /__w/1/s/src/coreclr/src/jit/phase.h:38
00007F87EFE41D60 00007F87F3104420 libclrjit.so!Compiler::compCompile(void**, unsigned int*, JitFlags*) + 2672 at /__w/1/s/src/coreclr/src/jit/phase.h:38
00007F87EFE42030 00007F87F3108065 libclrjit.so!Compiler::compCompileHelper(CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_METHOD_INFO*, void**, unsigned int*, JitFlags*) + 2581 at /__w/1/s/src/coreclr/src/jit/compiler.hpp:3947
00007F87EFE420F0 00007F87F3106AF7 libclrjit.so!Compiler::compCompile(CORINFO_METHOD_STRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_METHOD_INFO*, void**, unsigned int*, JitFlags*) + 2391 at /__w/1/s/src/coreclr/src/jit/compiler.cpp:5409
00007F87EFE420F0 00007F87F3106AD6 libclrjit.so!Compiler::compCompile(CORINFO_METHOD_STRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_METHOD_INFO*, void**, unsigned int*, JitFlags*) + 2358 at /__w/1/s/src/coreclr/src/jit/compiler.cpp:5409
00007F87EFE42170 00007F87F3108E22 libclrjit.so!jitNativeCode(CORINFO_METHOD_STRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_METHOD_INFO*, void**, unsigned int*, JitFlags*, void*) + 866 at /__w/1/s/src/coreclr/src/jit/compiler.cpp:6711
00007F87EFE42170 00007F87F3108C7C libclrjit.so!jitNativeCode(CORINFO_METHOD_STRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_METHOD_INFO*, void**, unsigned int*, JitFlags*, void*) + 444 at /__w/1/s/src/coreclr/src/jit/compiler.cpp:6668
00007F87EFE42170 00007F87F3108C78 libclrjit.so!jitNativeCode(CORINFO_METHOD_STRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_METHOD_INFO*, void**, unsigned int*, JitFlags*, void*) + 440 at /__w/1/s/src/coreclr/src/jit/compiler.cpp:6664
00007F87EFE42780 00007F87F31129D6 libclrjit.so!CILJit::compileMethod(ICorJitInfo*, CORINFO_METHOD_INFO*, unsigned int, unsigned char**, unsigned int*) + 278 at /__w/1/s/src/coreclr/src/jit/ee_il_dll.cpp:304
00007F87EFE42820 00007F87F45BBFF6 libcoreclr.so!invokeCompileMethodHelper(EEJitManager*, CEEInfo*, CORINFO_METHOD_INFO*, CORJIT_FLAGS, unsigned char**, unsigned int*) + 294 at /__w/1/s/src/coreclr/src/vm/jitinterface.cpp:12219
00007F87EFE42870 00007F87F45BC198 libcoreclr.so!CallCompileMethodWithSEHWrapper(EEJitManager*, CEEInfo*, CORINFO_METHOD_INFO*, CORJIT_FLAGS, unsigned char**, unsigned int*, NativeCodeVersion) + 136 at /__w/1/s/src/coreclr/src/vm/jitinterface.cpp:12284
00007F87EFE42870 00007F87F45BC167 libcoreclr.so!CallCompileMethodWithSEHWrapper(EEJitManager*, CEEInfo*, CORINFO_METHOD_INFO*, CORJIT_FLAGS, unsigned char**, unsigned int*, NativeCodeVersion) + 87 at /__w/1/s/src/coreclr/src/vm/jitinterface.cpp:12282
00007F87EFE42870 00007F87F45BC167 libcoreclr.so!CallCompileMethodWithSEHWrapper(EEJitManager*, CEEInfo*, CORINFO_METHOD_INFO*, CORJIT_FLAGS, unsigned char**, unsigned int*, NativeCodeVersion) + 87 at /__w/1/s/src/coreclr/src/vm/jitinterface.cpp:12282
00007F87EFE428F0 00007F87F45BD299 libcoreclr.so!UnsafeJitFunction(PrepareCodeConfig*, COR_ILMETHOD_DECODER*, CORJIT_FLAGS, unsigned int*) + 2713 at /__w/1/s/src/coreclr/src/vm/jitinterface.cpp:12859
00007F87EFE43490 00007F87F4604407 libcoreclr.so!MethodDesc::JitCompileCodeLocked(PrepareCodeConfig*, ListLockEntryBase<NativeCodeVersion>*, unsigned int*, CORJIT_FLAGS*) + 663 at /__w/1/s/src/coreclr/src/vm/prestub.cpp:1013
00007F87EFE435C0 00007F87F4603DB6 libcoreclr.so!MethodDesc::JitCompileCodeLockedEventWrapper(PrepareCodeConfig*, ListLockEntryBase<NativeCodeVersion>*) + 886 at /__w/1/s/src/coreclr/src/vm/prestub.cpp:866
00007F87EFE436B0 00007F87F4603486 libcoreclr.so!MethodDesc::JitCompileCode(PrepareCodeConfig*) + 966 at /__w/1/s/src/coreclr/src/vm/prestub.cpp:805
00007F87EFE43780 00007F87F4602965 libcoreclr.so!MethodDesc::PrepareILBasedCode(PrepareCodeConfig*) + 229 at /__w/1/s/src/coreclr/src/vm/prestub.cpp:404
00007F87EFE43810 00007F87F46027FA libcoreclr.so!MethodDesc::PrepareInitialCode(CallerGCMode) + 186 at /__w/1/s/src/coreclr/src/vm/prestub.cpp:0
00007F87EFE43810 00007F87F46027C0 libcoreclr.so!MethodDesc::PrepareInitialCode(CallerGCMode) + 128 at /__w/1/s/src/coreclr/src/vm/prestub.cpp:331
00007F87EFE43890 00007F87F488B0C1 libcoreclr.so!JitILStub(MethodDesc*) + 49 at /__w/1/s/src/coreclr/src/vm/dllimport.cpp:5950
00007F87EFE438B0 00007F87F488B31B libcoreclr.so!NDirect::GetStubForILStub(NDirectMethodDesc*, MethodDesc**, unsigned int) + 379 at /__w/1/s/src/coreclr/src/vm/dllimport.cpp:5892
00007F87EFE43920 00007F87F488BA1F libcoreclr.so!GetStubForInteropMethod(MethodDesc*, unsigned int, MethodDesc**) + 559 at /__w/1/s/src/coreclr/src/vm/dllimport.cpp:6083
00007F87EFE439C0 00007F87F4607ABE libcoreclr.so!MethodDesc::DoPrestub(MethodTable*, CallerGCMode) + 2062 at /__w/1/s/src/coreclr/src/vm/prestub.cpp:0
00007F87EFE43AE0 00007F87F4606CC3 libcoreclr.so!PreStubWorker + 2387 at /__w/1/s/src/coreclr/src/vm/prestub.cpp:1954
00007F87EFE43B10                  [PrestubMethodFrame: 00007f87efe43b10] System.Security.Cryptography.Algorithms.ni.dll!/root/helix/work/correlation/System.Security.Cryptography.Algorithms.ni.dll!Unknown
00007F87EFE43BF0 00007F87F47F7874 libcoreclr.so!ThePreStub + 92 at /__w/1/s/src/coreclr/src/vm/amd64/theprestubamd64.S:19
00007F87EFE43CE0 00007F877D16C01A System.Security.Cryptography.Algorithms.ni.dll!/root/helix/work/correlation/System.Security.Cryptography.Algorithms.ni.dll!Unknown + 26
00007F87EFE43D00 00007F877D16BD6E System.Security.Cryptography.Algorithms.ni.dll!/root/helix/work/correlation/System.Security.Cryptography.Algorithms.ni.dll!Unknown + 206
00007F87EFE43D30 00007F877D17F092 System.Security.Cryptography.Algorithms.ni.dll!/root/helix/work/correlation/System.Security.Cryptography.Algorithms.ni.dll!Unknown + 50
00007F87EFE43D50 00007F877D17F033 System.Security.Cryptography.Algorithms.ni.dll!/root/helix/work/correlation/System.Security.Cryptography.Algorithms.ni.dll!Unknown + 19
00007F87EFE43D60 00007F877BC0A1E1 Microsoft.Diagnostics.Tracing.TraceEvent.ni.dll!/root/helix/work/correlation/Microsoft.Diagnostics.Tracing.TraceEvent.ni.dll!Unknown + 353
00007F87EFE43DE0 00007F877BBE1F53 Microsoft.Diagnostics.Tracing.TraceEvent.ni.dll!/root/helix/work/correlation/Microsoft.Diagnostics.Tracing.TraceEvent.ni.dll!Unknown + 499
00007F87EFE43EA0 00007F877BBE17B9 Microsoft.Diagnostics.Tracing.TraceEvent.ni.dll!/root/helix/work/correlation/Microsoft.Diagnostics.Tracing.TraceEvent.ni.dll!Unknown + 217
00007F87EFE43EF0 00007F877BBDFEE8 Microsoft.Diagnostics.Tracing.TraceEvent.ni.dll!/root/helix/work/correlation/Microsoft.Diagnostics.Tracing.TraceEvent.ni.dll!Unknown + 200
00007F87EFE43F50 00007F877BBDFCC1 Microsoft.Diagnostics.Tracing.TraceEvent.ni.dll!/root/helix/work/correlation/Microsoft.Diagnostics.Tracing.TraceEvent.ni.dll!Unknown + 177
00007F87EFE44070 00007F877BBE1566 Microsoft.Diagnostics.Tracing.TraceEvent.ni.dll!/root/helix/work/correlation/Microsoft.Diagnostics.Tracing.TraceEvent.ni.dll!Unknown + 262
00007F87EFE440B0 00007F877BBE1294 Microsoft.Diagnostics.Tracing.TraceEvent.ni.dll!/root/helix/work/correlation/Microsoft.Diagnostics.Tracing.TraceEvent.ni.dll!Unknown + 260
00007F87EFE440E0 00007F877C151255 Microsoft.Diagnostics.FastSerialization.ni.dll!/root/helix/work/correlation/Microsoft.Diagnostics.FastSerialization.ni.dll!Unknown + 549
00007F87EFE44130 00007F877C150540 Microsoft.Diagnostics.FastSerialization.ni.dll!/root/helix/work/correlation/Microsoft.Diagnostics.FastSerialization.ni.dll!Unknown + 560
00007F87EFE44170 00007F877BBDF9D8 Microsoft.Diagnostics.Tracing.TraceEvent.ni.dll!/root/helix/work/correlation/Microsoft.Diagnostics.Tracing.TraceEvent.ni.dll!Unknown + 56
00007F87EFE441A0 00007F877B614667 eventpipe.dll!/root/helix/work/workitem/profiler/eventpipe/eventpipe/eventpipe.dll!Unknown + 23 [/Users/runner/runners/2.166.4/work/1/s/src/coreclr/tests/src/profiler/eventpipe/eventpipe.cs @ 87]
00007F87EFE441B0 00007F877AE8346D System.Private.CoreLib.dll!/root/helix/work/correlation/System.Private.CoreLib.dll!Unknown + 205
00007F87EFE441E0 00007F877AE8E401 System.Private.CoreLib.dll!/root/helix/work/correlation/System.Private.CoreLib.dll!Unknown + 129
00007F87EFE44230 00007F877AE83612 System.Private.CoreLib.dll!/root/helix/work/correlation/System.Private.CoreLib.dll!Unknown + 114
00007F87EFE44260 00007F87F47F6B4F libcoreclr.so!CallDescrWorkerInternal + 124 at /__w/1/s/src/coreclr/src/pal/inc/unixasmmacrosamd64.inc:845
00007F87EFE44280 00007F87F46858EC libcoreclr.so!CallDescrWorkerWithHandler(CallDescrData*, int) + 220 at /__w/1/s/src/coreclr/src/pal/inc/pal.h:4607
00007F87EFE442D0 00007F87F4686440 libcoreclr.so!MethodDescCallSite::CallTargetWorker(unsigned long const*, unsigned long*, int) + 2000 at /__w/1/s/src/coreclr/src/vm/callhelpers.cpp:546
00007F87EFE444C0 00007F87F4694628 libcoreclr.so!ThreadNative::KickOffThread_Worker(void*) + 1176 at /__w/1/s/src/coreclr/src/vm/callhelpers.h:0
00007F87EFE444C0 00007F87F469461F libcoreclr.so!ThreadNative::KickOffThread_Worker(void*) + 1167 at /__w/1/s/src/coreclr/src/vm/callhelpers.h:459
00007F87EFE44670 00007F87F4640F6C libcoreclr.so!ManagedThreadBase_DispatchOuter(ManagedThreadCallState*) + 524 at /__w/1/s/src/coreclr/src/vm/threads.cpp:0
00007F87EFE44670 00007F87F4640F65 libcoreclr.so!ManagedThreadBase_DispatchOuter(ManagedThreadCallState*) + 517 at /__w/1/s/src/coreclr/src/vm/threads.cpp:7333
00007F87EFE44670 00007F87F4640E99 libcoreclr.so!ManagedThreadBase_DispatchOuter(ManagedThreadCallState*) + 313 at /__w/1/s/src/coreclr/src/vm/threads.inl:38
00007F87EFE44670 00007F87F4640E95 libcoreclr.so!ManagedThreadBase_DispatchOuter(ManagedThreadCallState*) + 309 at /__w/1/s/src/coreclr/src/vm/threads.cpp:7536
00007F87EFE44670 00007F87F4640E1B libcoreclr.so!ManagedThreadBase_DispatchOuter(ManagedThreadCallState*) + 187 at /__w/1/s/src/coreclr/src/vm/threads.cpp:7534
00007F87EFE446A8                  [DebuggerU2MCatchHandlerFrame: 00007f87efe446a8]
00007F87EFE447D0 00007F87F464172D libcoreclr.so!ManagedThreadBase::KickOff(void (*)(void*), void*) + 45 at /__w/1/s/src/coreclr/src/vm/threads.cpp:0
00007F87EFE447D0 00007F87F4641715 libcoreclr.so!ManagedThreadBase::KickOff(void (*)(void*), void*) + 21 at /__w/1/s/src/coreclr/src/vm/threads.cpp:7307
00007F87EFE44800 00007F87F46948B6 libcoreclr.so!ThreadNative::KickOffThread(void*) + 278 at /__w/1/s/src/coreclr/src/vm/comsynchronizable.cpp:328
00007F87EFE44880 00007F87F4B6BF31 libcoreclr.so!CorUnix::CPalThread::ThreadEntry(void*) + 641 at /__w/1/s/src/coreclr/src/pal/src/thread/thread.cpp:1795
```
I'm not positive that this thread is problematic, but I do see the JIT morphing the AST and injecting assertion related nodes.  @BruceForstall is this indicative of an issue being encountered?

As I said earlier, most other threads show up as truncated to the musl linker so there isn't much information that can be gleaned easily:
```
(lldb) bt all
  thread #1, name = 'corerun', stop reason = signal SIGABRT
    frame #0: 0x00007f87f531447a ld-musl-x86_64.so.1
  thread #2, stop reason = signal SIGABRT
    frame #0: 0x00007f87f52f51ae ld-musl-x86_64.so.1`syscall + 158
  thread #3, stop reason = signal SIGABRT
    frame #0: 0x00007f87f52f51ae ld-musl-x86_64.so.1`syscall + 158
  thread #4, stop reason = signal SIGABRT
    frame #0: 0x00007f87f531447a ld-musl-x86_64.so.1
  thread #5, stop reason = signal SIGABRT
    frame #0: 0x00007f87f531447a ld-musl-x86_64.so.1
  thread #6, stop reason = signal SIGABRT
    frame #0: 0x00007f87f531447a ld-musl-x86_64.so.1
  thread #7, stop reason = signal SIGABRT
    frame #0: 0x00007f87f531447a ld-musl-x86_64.so.1
  thread #8, stop reason = signal SIGABRT
    frame #0: 0x00007f87f531447a ld-musl-x86_64.so.1
  thread #9, stop reason = signal SIGABRT
    frame #0: 0x00007f87f531447a ld-musl-x86_64.so.1
* thread #10, stop reason = signal SIGABRT
  * frame #0: 0x00007f87f32b1480 libclrjit.so`ValueNumStore::VNNormalValue(this=0x0000000000000000, vnp=<unavailable>, vnk=VNK_Conservative) at valuenum.cpp:1372
    frame #1: 0x00007f87f30d16af libclrjit.so`Compiler::optCreateAssertion(GenTree*, GenTree*, Compiler::optAssertionKind, bool) [inlined] ValueNumStore::VNConservativeNormalValue(this=<unavailable>, vnp=<unavailable>) at valuenum.h:425:16
    frame #2: 0x00007f87f30d16a5 libclrjit.so`Compiler::optCreateAssertion(this=<unavailable>, op1=0x000055af9bd0c7b8, op2=<unavailable>, assertionKind=<unavailable>, helperCallArgs=<unavailable>) at assertionprop.cpp:1032
    frame #3: 0x00007f87f30d36d3 libclrjit.so`Compiler::optAssertionGen(this=0x000055af9bd0abc8, tree=0x000055af9bd0c828) at assertionprop.cpp:0:65
    frame #4: 0x00007f87f3240e39 libclrjit.so`Compiler::fgMorphTreeDone(this=0x000055af9bd0abc8, tree=0x000055af9bd0c828, oldTree=<unavailable>, morphNum=<unavailable>) at morph.cpp:15442:5
...
```
My guess is one of those threads is the Diagnostics Server, but it was put in a bad state by the segfault and the dump was taken at the ASSERT so we can't see anything.


The following threads were clrthreads:
```
(lldb) clrthreads
ThreadCount:      4
UnstartedThread:  0
BackgroundThread: 2
PendingThread:    0
DeadThread:       0
Hosted Runtime:   no
                                                                                                        Lock
 DBG   ID OSID ThreadOBJ           State GC Mode     GC Alloc Context                  Domain           Count Apt Exception
   1    1   3a 000055AF9BC2AD50    20020 Preemptive  00007F875A8D77D0:00007F875A8D9008 000055AF9BBEA0F0 -00001 Ukn
   8    2   41 000055AF9BC83270    21220 Preemptive  00007F875A8C9C10:00007F875A8CAAD8 000055AF9BBEA0F0 -00001 Ukn (Finalizer)
  10    4   47 000055AF9BFB1CD0    21020 Preemptive  00007F875A8D9098:00007F875A8DAFD0 000055AF9BBEA0F0 -00001 Ukn
   9    5   45 000055AF9BFBCAF0    20220 Preemptive  0000000000000000:0000000000000000 000055AF9BBEA0F0 -00001 Ukn
```

As far as I can tell, the segfault from the logs did not make it into the coredump in diagnosable fashion.  This is understandable as it looks like the dump was triggered by the `ASSERT` rather than the original segfault due to the try/catch logic.  I can't seem to locate the `ASSERT` that should have been at the top of the stack for one of the threads, though.  My guess is the thread that encountered the segfault was put in a non-recoverable state and since the dump only happened at the `ASSERT` we got things in a bad state.  It is also possible I'm not seeing the whole picture because the dump was collected incorrectly and it's missing the information necessary to build a stack back from the linker.

**Comment by josalem** [comment]
CC @jashook

**Comment by v-haren** [comment]
failed again in job: [runtime-coreclr jitstress2-jitstressregs 20200621.1 ](https://dev.azure.com/dnceng/public/_build/results?buildId=697815&view=ms.vss-test-web.build-test-results-tab&runId=21625482&resultId=100005&paneView=debug)

failed test: profiler/eventpipe/eventpipe/eventpipe.sh


Error message
~~~
Assert failure(PID 19650 [0x00004cc2], Thread: 4170717 [0x3fa3dd]): !"Hit an error in the diagnostic server thread
."
 File: /Users/runner/runners/2.170.1/work/1/s/src/coreclr/src/vm/diagnosticserver.cpp Line: 104
 Image: /private/tmp/helix/working/C4440A6D/p/corerun

task_for_pid(19650) FAILED 5 (os/kern) failure
Unhandled exception. System.Exception: Profiler tests are expected to contain the text 'PROFILER TEST PASSES' in the console output of the profilee app to indicate a passing test. Usually it is printed from the Shutdown() method of the profiler implementation. This text was not found in the output above.
 at Profiler.Tests.ProfilerTestRunner.LogTestFailure(String error) in /Users/runner/runners/2.170.1/work/1/s/src/coreclr/tests/src/profiler/common/ProfilerTestRunner.cs:line 137
 at Profiler.Tests.ProfilerTestRunner.Run(String profileePath, String testName, Guid profilerClsid, String profileeArguments, ProfileeOptions profileeOptions) in /Users/runner/runners/2.170.1/work/1/s/src/coreclr/tests/src/profiler/common/ProfilerTestRunner.cs:line 91
 at EventPipeTests.EventPipe.Main(String[] args) in /Users/runner/runners/2.170.1/work/1/s/src/coreclr/tests/src/profiler/eventpipe/eventpipe.cs:line 32
task_for_pid(19649) FAILED 5 (os/kern) failure
/private/tmp/helix/working/C4440A6D/w/B2E40925/e/profiler/eventpipe/eventpipe/eventpipe.sh: line 317: 19649 Abort trap: 6 (core dumped) $LAUNCHER $ExePath "${CLRTestExecutionArguments[@]}"

Return code: 1
Raw output file: /private/tmp/helix/working/C4440A6D/w/B2E40925/e/profiler/eventpipe/Reports/profiler.eventpipe/eventpipe/eventpipe.output.txt
Raw output:
BEGIN EXECUTION
/tmp/helix/working/C4440A6D/p/corerun eventpipe.dll ''
Profiler path: /private/tmp/helix/working/C4440A6D/w/B2E40925/e/profiler/eventpipe/eventpipe/libProfiler.dylib
Profiler path: /private/tmp/helix/working/C4440A6D/w/B2E40925/e/profiler/eventpipe/eventpipe/libProfiler.dylib
Profiler.dll!DllGetClassObject
Profiler.dll!Profiler::Initialize
This method being jitted should trigger events firing...
TriggerMethod found! Sending event...
Writing minidump to file /cores/coredump.19650.dmp

Test failed: Profiler tests are expected to contain the text 'PROFILER TEST PASSES' in the console output of the profilee app to indicate a passing test. Usually it is printed from the Shutdown() method of the profiler implementation. This text was not found in the output above.
Writing minidump to file /cores/coredump.19649.dmp
Expected: 100
Actual: 134
END EXECUTION - FAILED

Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=/tmp/helix/working/C4440A6D/p
> /private/tmp/helix/working/C4440A6D/w/B2E40925/e/profiler/eventpipe/eventpipe/eventpipe.sh
Expected: True
Actual: False


Stack trace
   at profiler_eventpipe._eventpipe_eventpipe_._eventpipe_eventpipe_sh() in /Users/runner/runners/2.170.1/work/1/s/artifacts/tests/coreclr/OSX.x64.Checked/TestWrappers/profiler.eventpipe/profiler.eventpipe.XUnitWrapper.cs:line 140
~~~

**Comment by v-haren** [comment]
also failed in job: [runtime-coreclr gcstress0x3-gcstress0xc 20200719.1 ](https://dev.azure.com/dnceng/public/_build/results?buildId=737239&view=ms.vss-test-web.build-test-results-tab&runId=22831294&resultId=100009&paneView=debug)

failed test: profiler/eventpipe/eventpipe/eventpipe.sh


Error message
~~~
Unhandled exception. System.Exception: Profiler tests are expected to contain the text 'PROFILER TEST PASSES' in the console output of the profilee app to indicate a passing test. Usually it is printed from the Shutdown() method of the profiler implementation. This text was not found in the output above.
   at Profiler.Tests.ProfilerTestRunner.LogTestFailure(String error) in /Users/runner/work/1/s/src/tests/profiler/common/ProfilerTestRunner.cs:line 136
   at Profiler.Tests.ProfilerTestRunner.Run(String profileePath, String testName, Guid profilerClsid, String profileeArguments, ProfileeOptions profileeOptions) in /Users/runner/work/1/s/src/tests/profiler/common/ProfilerTestRunner.cs:line 90
   at EventPipeTests.EventPipe.Main(String[] args) in /Users/runner/work/1/s/src/tests/profiler/eventpipe/eventpipe.cs:line 31
apply_reg_state: ip and cfa unchanged; stopping here (ip=0x7fb2d54be0)
/root/helix/work/workitem/profiler/eventpipe/eventpipe/eventpipe.sh: line 333:   153 Aborted                 $LAUNCHER $ExePath "${CLRTestExecutionArguments[@]}"

Return code:      1
Raw output file:      /root/helix/work/workitem/profiler/eventpipe/Reports/profiler.eventpipe/eventpipe/eventpipe.output.txt
Raw output:
BEGIN EXECUTION
/root/helix/work/correlation/corerun eventpipe.dll ''
Profiler path: /root/helix/work/workitem/profiler/eventpipe/eventpipe/libProfiler.so
Profiler path: /root/helix/work/workitem/profiler/eventpipe/eventpipe/libProfiler.so
Profiler.dll!DllGetClassObject
Profiler.dll!Profiler::Initialize
This method being jitted should trigger events firing...
SimpleEvent got an out of order event expected=9999 actual=9984.
Test validation failed (EventPipeClient.exe)
    success=False
    allTypesEventCount=1 
    arrayTypeEventCount=1 
    emptyEventCount=10
    simpleEventCount=10000
TriggerMethod found! Sending event...
Profiler.dll!Profiler::Shutdown
PROFILER TEST PASSES

Test failed: Profiler tests are expected to contain the text 'PROFILER TEST PASSES' in the console output of the profilee app to indicate a passing test. Usually it is printed from the Shutdown() method of the profiler implementation. This text was not found in the output above.
Writing minidump to file /home/helixbot/dotnetbuild/dumps/coredump.153.dmp
Process 153 corerun
Written 119607296 bytes (29201 pages) to core file
Dump successfully written
Expected: 100
Actual: 134
END EXECUTION - FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=/root/helix/work/correlation
> /root/helix/work/workitem/profiler/eventpipe/eventpipe/eventpipe.sh
Expected: True
Actual:   False


Stack trace
   at profiler_eventpipe._eventpipe_eventpipe_._eventpipe_eventpipe_sh() in /__w/1/s/artifacts/tests/coreclr/Linux.arm64.Checked/TestWrappers/profiler.eventpipe/profiler.eventpipe.XUnitWrapper.cs:line 140
~~~

**Comment by josalem** [comment]
@davmason this test failure looks different than the ones before.  It looks like the test passed, but the harness didn't observe the `PROFILER TEST PASSES` in the test output.  Any ideas?

**Comment by davmason** [comment]
@josalem The wording is bad. The profiler test runner expects the app to return the magic exit code of 100 from the managed part and the output to contain the PROFILER TEST PASSES from the native profiler part.

What happened here is the native profiler did everything and wrote PROFILER TEST PASSES but the managed bit that verifies the bit saw an out of order event.

That's a weird failure. It writes 10,000 events that each have a payload with their event order (i.e. first event payload=0, second event payload=1, etc). And the first 9,999 came through ok but number 10,000 had a value of 9,984.

**Comment by tommcdon** [comment]
Moving to 6.0.  Out of order events with GC stress enabled needs further investigation.

**Comment by v-haren** [comment]
failed again in job: [runtime-coreclr jitstress 20200729.1 ](https://dev.azure.com/dnceng/public/_build/results?buildId=750921&view=ms.vss-test-web.build-test-results-tab&runId=23285924&resultId=100588&paneView=debug)

failed test: profiler/elt/slowpatheltleave/slowpatheltleave.sh


Error message
~~~
Assert failure(PID 169 [0x000000a9], Thread: 169 [0x00a9]): hfaFieldSize == 8
    File: /__w/1/s/src/coreclr/src/vm/arm64/profiler.cpp Line: 265
    Image: /root/helix/work/correlation/corerun

apply_reg_state: ip and cfa unchanged; stopping here (ip=0x7f9a23ba30)
apply_reg_state: ip and cfa unchanged; stopping here (ip=0x7f9a23ba30)
apply_reg_state: ip and cfa unchanged; stopping here (ip=0x7f9a230470)
Unhandled exception. System.Exception: Profiler tests are expected to contain the text 'PROFILER TEST PASSES' in the console output of the profilee app to indicate a passing test. Usually it is printed from the Shutdown() method of the profiler implementation. This text was not found in the output above.
   at Profiler.Tests.ProfilerTestRunner.LogTestFailure(String error) in /Users/runner/work/1/s/src/tests/profiler/common/ProfilerTestRunner.cs:line 138
   at Profiler.Tests.ProfilerTestRunner.Run(String profileePath, String testName, Guid profilerClsid, String profileeArguments, ProfileeOptions profileeOptions) in /Users/runner/work/1/s/src/tests/profiler/common/ProfilerTestRunner.cs:line 92
   at SlowPathELTTests.SlowPathELTLeave.Main(String[] args) in /Users/runner/work/1/s/src/tests/profiler/elt/slowpatheltleave.cs:line 29
apply_reg_state: ip and cfa unchanged; stopping here (ip=0x7f78bbe470)
/root/helix/work/workitem/profiler/elt/slowpatheltleave/slowpatheltleave.sh: line 347:   153 Aborted                 $LAUNCHER $ExePath "${CLRTestExecutionArguments[@]}"

Return code:      1
Raw output file:      /root/helix/work/workitem/profiler/elt/Reports/profiler.elt/slowpatheltleave/slowpatheltleave.output.txt
Raw output:
BEGIN EXECUTION
/root/helix/work/correlation/corerun slowpatheltleave.dll ''
Profiler path: /root/helix/work/workitem/profiler/elt/slowpatheltleave/libProfiler.so
Profiler.dll!DllGetClassObject
Profiler.dll!Profiler::Initialize
Testing leave hooks
Writing minidump with heap to file /home/helixbot/dotnetbuild/dumps/coredump.169.dmp
Process 169 corerun
Written 63209472 bytes (15432 pages) to core file
Dump successfully written

Test failed: Profiler tests are expected to contain the text 'PROFILER TEST PASSES' in the console output of the profilee app to indicate a passing test. Usually it is printed from the Shutdown() method of the profiler implementation. This text was not found in the output above.
Writing minidump with heap to file /home/helixbot/dotnetbuild/dumps/coredump.153.dmp
Process 153 corerun
Written 115212288 bytes (28128 pages) to core file
Dump successfully written
Expected: 100
Actual: 134
END EXECUTION - FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=/root/helix/work/correlation
> /root/helix/work/workitem/profiler/elt/slowpatheltleave/slowpatheltleave.sh
Expected: True
Actual:   False


Stack trace
   at profiler_elt._slowpatheltleave_slowpatheltleave_._slowpatheltleave_slowpatheltleave_sh() in /__w/1/s/artifacts/tests/coreclr/Linux.arm64.Checked/TestWrappers/profiler.elt/profiler.elt.XUnitWrapper.cs:line 237
~~~

**Comment by davmason** [comment]
@AndyAyersMS is this failure what you expected to see when the ELT tests were checked in? [runtime-coreclr jitstress 20200729.1 ](https://dev.azure.com/dnceng/public/_build/results?buildId=750921&view=ms.vss-test-web.build-test-results-tab&runId=23285924&resultId=100588&paneView=debug)

**Comment by BruceForstall** [comment]
The slowpatheltleave issue is a separate issue. I've moved it to https://github.com/dotnet/runtime/issues/40277.

**Comment by davmason** [comment]
@josalem @sywhang I think the original failure here of "Hit an error in the diagnostic server thread" is a dupe of #39669 (the filter data use after free bug). Looking at how the code used to be, it would do an EX_CATCH and then assert. When it was changed to not catch exceptions in debug builds is when we started seeing the real call stack.

Closing this as a dupe of #39669.
